third_party/ienn/src/ie_compilation.cpp

```diff
@@ -359,155 +316,166 @@ int32_t Compilation::AddConvolution(const Operation& operation) {
+      if (params.depthwise_multiplier != 1) {
+        std::cout << "depthwise_multiplier " << params.depthwise_multiplier
+                  << " is not supported";
+        return error_t::BAD_DATA;
       }
-    }
-    layer_id_map_[output_index] = layer_id;
-  } catch (const std::exception& ex) {
-    std::cout << "[IE] failed to add convolution layer " << ex.what();
-    return error_t::OP_FAILED;
-  }
-  return error_t::NOT_ERROR;
-}
 
-int32_t Compilation::AddPooling(const Operation& operation) {
-  // Setup pooling params.
-  PoolingParams params;
-  int32_t result = GetPoolingParams(model_, operation, params);
-  if (result != error_t::NOT_ERROR)
-    return result;
-  const uint32_t input_index = operation.inputs[0];
-  if (layer_id_map_.find(input_index) == layer_id_map_.end()) {
-    std::cout << "The layer for operand index " << input_index
-              << " is not ready";
-    return error_t::BAD_DATA;
-  }
-  try {
-    const uint32_t output_index = operation.outputs[0];
-    std::string output_name(std::to_string(output_index));
-    std::string name(output_name);
-    if (params.fuse_code != FUSED_NONE) {
-      name = name + "_pre_fuse";
-    }
-    const size_t input_layer_id = layer_id_map_[input_index];
-    size_t layer_id = builder_->addLayer(
-        {{input_layer_id}},
-        Builder::PoolingLayer(name)
-            .setExcludePad(true)
-            .setKernel({params.filter_height, params.filter_width})
-            .setStrides({params.stride_width, params.stride_height})
-            .setPaddingsBegin({params.padding_top, params.padding_left})
-            .setPaddingsEnd({params.padding_bottom, params.padding_right})
-            .setPoolingType(operation.type == AVERAGE_POOL_2D
-                                ? Builder::PoolingLayer::PoolingType::AVG
-                                : Builder::PoolingLayer::PoolingType::MAX)
-            .setRoundingType(Builder::PoolingLayer::RoundingType::FLOOR));
-    if (params.fuse_code != FUSED_NONE) {
-      result = AddActivationByFusedCode(params.fuse_code, layer_id, output_name,
-                                        layer_id);
+      if (params.output_channel !=
+          params.input_channel * params.depthwise_multiplier) {
+        std::cout
+            << "Failed assertion: outputFeatureChannels "
+            << params.output_channel
+            << " in IE depthwise convolution descriptor must be multiplie of "
+               "inFeatureChannels "
+            << params.input_channel;
+        return error_t::BAD_DATA;
+      }
+      Blob::Ptr blob;
+      result = CreateBlob(weights_index, blob);
       if (result != error_t::NOT_ERROR) {
         return result;
       }
+      SizeVector weights_dims{params.depth_out, 1, 1, params.filter_height,
+                              params.filter_width};
+      auto weights_node = std::make_shared<op::Constant>(
+          element::f32, Shape(weights_dims), blob->buffer().as<float*>());
```

Comment:
It will be failed on Myriad plugin because it only accepts FP16 data.
And We need verify test case and examples on all hardware including CPU/GPU/GNA/VPU.